tuf/client/updater.py

```diff
@@ -694,6 +713,22 @@ def __init__(self, repository_name, repository_mirrors):
     # Save the validated arguments.
     self.repository_name = repository_name
     self.mirrors = repository_mirrors
+    self.targets_map_file = None
+
+    # If targets_map_filename is provided, load the targets_map_file.
+    if targets_map_filename is not None:
+      securesystemslib.formats.PATH_SCHEMA.check_match(targets_map_filename)
+
+      try:
+        self.targets_map_file = securesystemslib.util.load_json_file(
+            targets_map_filename)
+
+      except (securesystemslib.exceptions.Error) as e:
+        raise tuf.exceptions.Error('Cannot load the targets map file: ' + str(e))
```

Comment:
Please avoid generic exceptions in public API if possible.

Also document the exception (alternatively remove the 'Exceptions' section from the docs if it's known to be outdated).

Reply:
Do you think a FormatError would make more sense here (to indicate that the targets map file specified an invalid filename)? Or something else?

Reply:
Good question: securesystemslib seems to have a wide variety of errors here (some of which clearly aren't formaterrors) so maybe you can't avoid a generic one. In that case at least document the exceptions that may be raised.

By the way, the check_match() call just before this seems to not be needed as securesystemslib does that (unless the point was to trigger the FormatError specifically before the try-except)